riscv/triggers.cc

```diff
@@ -296,7 +296,10 @@ std::optional<match_result_t> icount_t::detect_icount_match(processor_t * const
   if (pending) {
     pending = 0;
     hit = true;
-    ret = match_result_t(TIMING_BEFORE, action);
+    if (action == ACTION_DEBUG_EXCEPTION) // decrease count only on firing breakpoint exception action
+      ret = match_result_t(TIMING_BEFORE, ACTION_DEBUG_EXCEPTION);
+    else
+      return match_result_t(TIMING_BEFORE, action);
```

Comment:
I don't like the duplicate expressions here in the if and else clauses. How about set `ret` first (the existing code), then test the action and `return ret` if needed?

Reply:
The if-else clauses here could be more transparent. It messes up the execution flow and exits the function early, reducing the code's readability. 

How about simplifying the function by keeping the function always goes through three steps: 1) setting ret=action, 2) decreasing counter, and 3) return ret. Although this implementation has a more complicated checking on whether to decrease the counter, it provides a better overall readability for future maintainers.